Aquarius/Src/Aquarius/Sound/Devices.cpp

```diff
@@ -0,0 +1,76 @@
+#include "Devices.h"
+#include "Aquarius/Core/Log.h"
+#include <AL/alc.h>
+
+
+namespace Aquarius {
+
+    namespace Sound {
+
+        Device::Device()
+        {
+            Initialize();
+        }
+
+        void Device::Initialize()
+        {
+            m_Device = alcOpenDevice(nullptr);
+
+            if (!m_Device)
+            {
+                AQ_CORE_ERROR("Failed to open a sound device");
+            }
+
+            else
+            {
+                AQ_CORE_INFO("A sound device was successfully opened");
+            }
+
+            m_Context = alcCreateContext(m_Device, nullptr);
+
+            if (!m_Context) {
+                AQ_CORE_ERROR("Failed to create the sound context");
+            } else {
+                AQ_CORE_INFO("Sound context created successfully");
+            }
+
+            if (!alcMakeContextCurrent(m_Context))
+            {
+                AQ_CORE_ERROR("Failed to make the context current");
+            }
+        }
+
+        Device * Device::get()
+        {
+            static Device* device = new Device();
```

Comment:
This doesn't get deleted anywhere, I'm wondering if this could cause a memory leak